### PR TITLE
feat: add declaration file to prevent type error

### PR DESCRIPTION
**Pull request by ChrisCho-H**
I know it's pure js project, but a lot of typescript dev utilizes this module to optimize UTXO selection for bitcoin transaction. Just adding simple declaration file will make this library much more usable for typescript dev!

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    value: number,
+    nonWitnessUtxo? : Buffer,
+    witnessUtxo? : {
+        script: Buffer,
+        value: number
+    }
+}
+export interface Target {
+    address: string,
+    value: number
```
Comment (Overtorment) [inline index.d.ts:13]: `value`  should be optional (in this case, library assings all remaining funds to this target)

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+}
+export interface SelectedUTXO {
+    inputs: UTXO[],
+    outputs: Target[],
```
Comment (Overtorment) [inline index.d.ts:17]: inputs & outputs are optional, this means lib failed to satisfy the conditions

**Review by Overtorment — CHANGES_REQUESTED** [review]

**Comment by ChrisCho-H** [comment]
Thx, I updated following your suggestion.

**Review by Overtorment — APPROVED** [review]
image
image

**Comment by junderw** [comment]
@Overtorment I gave you write permissions. Also, what is your npm user name?

**Review by junderw — APPROVED** [review]
Defer to @Overtorment

**Comment by LeeKorbisCa** [comment]
Thank you everyone for getting this in :heart:

**Comment by junderw** [comment]
... shouldn't this d.ts file be referenced in package.json?

**Comment by LeeKorbisCa** [comment]
Yes, in the types property, as you probably read.

**Comment by ChrisCho-H** [comment]
add types in package.json in 683e6dfae46795568172a4287260ce516b38eb7b

**Comment by junderw** [comment]
@Overtorment merge when ready.

**Comment by Overtorment** [comment]
@junderw i dont have npm account. dont trust them

**Comment by ChrisCho-H** [comment]
I made PR https://github.com/bitcoinjs/coinselect/pull/82 to calculate fee more precisely(varying on type of input/output), which also adds `script` field type for UTXO and Target(coinselect accepts script as input and output. I should have added in this PR actually), and `isTaproot` field type for UTXO to calculate p2tr input bytes.

**Comment by griga** [comment]
So this type declarations won't be published to npm ?

**Comment by ChrisCho-H** [comment]
@griga If u need typed one in npm u can use [my forked version](https://github.com/ChrisCho-H/bitcoinselect). It also supports segwit fee calculation.

**Pull request merged**